security/experimental_authenticators.rst

```diff
@@ -0,0 +1,432 @@
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/security
+                https://symfony.com/schema/dic/security/security-1.0.xsd">
+
+            <config enable-authenticator-manager="true">
+                <!-- ... -->
+
+                <!-- entry-point: configure the form authentication as the entry
+                                  point for unauthenticated users -->
+                <firewall name="main"
+                    entry-point="form_login"
+                >
+                    <!-- allow authentication using a form or HTTP basic -->
+                    <form-login/>
+                    <http-basic/>
+            </config>
+        </srv:container>
+
+    .. code-block:: php
+
+        // config/packages/security.php
+        use Symfony\Component\Security\Http\Firewall\AccessListener;
+
+        $container->loadFromExtension('security', [
+            'enable_authenticator_manager' => true,
+
+            // ...
+            'firewalls' => [
+                'main' => [
+                    // allow authentication using a form or HTTP basic
+                    'form_login' => null,
+                    'http_basic' => null,
+
+                    // configure the form authentication as the entry point for unauthenticated users
+                    'entry_point' => 'form_login'
+                ],
+            ],
+        ]);
+
+.. note::
+
+    You can also create your own authentication entry point by creating a
+    class that implements
+    :class:`Symfony\\Component\\Security\\Http\\EntryPoint\\AuthenticationEntryPointInterface`.
+    You can then set ``entry_point`` to the FQCN (e.g.
+    ``entry_point: App\Security\CustomEntryPoint``)
+
+.. _authenticators-removed-authentication-providers:
+
+Creating a Custom Authenticator
+-------------------------------
+
+Security traditionally could be extended by writing
+:doc:`custom authentication providers </security/custom_authentication_provider>`.
+The authenticator-based system dropped support for these providers and
+introduced a new authenticator interface as a base for custom
+authentication methods.
+
+.. tip::
+
+    :doc:`Guard authenticators </security/guard_authentication>` are still
+    supported in the authenticator-based system. It is however recommended
+    to also update these when you're refactoring your application to the
+    new system.
+
+    The new authenticator interface has quite some overlap with the guard
+    authenticator interface, making the rewrite easier. To avoid duplicated
+    class names, it is recommended to remove the ``symfony/security-guard``
+    package before rewriting your Security classes:
+
+    .. code-block:: terminal
+
+        $ composer remove symfony/security-guard
+
+Authenticators should implement the
+:class:`Symfony\\Component\\Security\\Http\\Authenticator\\AuthenticatorInterface`.
+The easiest way to use it is by extending
```

Comment:
What about « An alternative could be to extends the abstract ... »

Because an abstract is not always easier imo, sometimes implementing fully is easier :)